### PR TITLE
test: add codespell github action

### PR DESCRIPTION
**Pull request by andyzhangx**

### CODE REVIEW

**Comment by andyzhangx** [comment]
```
Error: ./NOTICE:4428: Unknwon ==> Unknown
Error: ./NOTICE:19511: Unknwon ==> Unknown
Error: ./notices_fix.sh:13: dependecies ==> dependencies
Error: ./notices_fix.sh:92: extnesion ==> extension
Error: ./notices_fix.sh:98: extnesion ==> extension
Error: ./README.md:135: suppored ==> supported
Error: ./setup/baseConfig.yaml:3: backgound ==> background
Error: ./setup/baseConfig.yaml:31: existant ==> existent
Error: ./setup/baseConfig.yaml:54: delted ==> deleted
Error: ./cmd/mount_all.go:163: Configue ==> Configure
Error: ./cmd/mountv1.go:174: supercede ==> supersede
Error: ./cmd/root.go:150: assosiated ==> associated
Error: ./cmd/mount.go:108: doesnt ==> doesn't, does not
Error: ./cmd/mount.go:275: initiliazing ==> initializing
Error: ./test/e2e_tests/file_test.go:509: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:520: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:521: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:524: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:529: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:534: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:538: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:539: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:542: fileTest ==> file test
Error: ./test/e2e_tests/file_test.go:545: fileTest ==> file test
Error: ./test/scripts/test_cases.csv:2: fo ==> of, for
Error: ./component/file_cache/lru_policy.go:68: immediatly ==> immediately
Error: ./component/file_cache/lru_policy.go:284: occured ==> occurred
Error: ./component/file_cache/file_cache_test.go:581: hasnt ==> hasn't
Error: ./component/file_cache/file_cache_test.go:995: untill ==> until
Error: ./component/file_cache/file_cache.go:272: doesnt ==> doesn't, does not
Error: ./component/file_cache/file_cache.go:440: retreived ==> retrieved
Error: ./component/file_cache/file_cache.go:445: retreived ==> retrieved
Error: ./component/file_cache/file_cache.go:456: ony ==> only
Error: ./component/file_cache/file_cache.go:522: non-existant ==> non-existent
Error: ./component/file_cache/file_cache.go:537: seperate ==> separate
Error: ./component/file_cache/file_cache.go:581: immediatly ==> immediately
Error: ./component/file_cache/file_cache.go:665: immediatly ==> immediately
Error: ./component/azstorage/datalake_test.go:115: doesnt ==> doesn't, does not
Error: ./component/azstorage/azauthspn.go:104: retreived ==> retrieved
Error: ./component/azstorage/azauthspn.go:136: retreived ==> retrieved
Error: ./component/azstorage/utils.go:362: extenstion ==> extension
Error: ./component/azstorage/utils.go:420: extention ==> extension
Error: ./component/azstorage/utils.go:420: retreive ==> retrieve
Error: ./component/azstorage/utils.go:437: henc ==> hence
Error: ./component/azstorage/block_blob_test.go:227: doesnt ==> doesn't, does not
Error: ./component/azstorage/utils_test.go:22: dum ==> dumb
Error: ./component/azstorage/datalake.go:128: retreive ==> retrieve
Error: ./component/azstorage/datalake.go:351: Retreive ==> Retrieve
Error: ./component/azstorage/azauthmsi.go:91: retreived ==> retrieved
Error: ./component/azstorage/azauthmsi.go:122: retreived ==> retrieved
Error: ./component/azstorage/azauthkey.go:58: autheticate ==> authenticate
Error: ./component/azstorage/azauthkey.go:80: autheticate ==> authenticate
Error: ./component/azstorage/config.go:340: doesnt ==> doesn't, does not
Error: ./component/azstorage/config.go:373: doesnt ==> doesn't, does not
Error: ./component/azstorage/azstorage.go:147: Intialize ==> Initialize
Error: ./component/azstorage/azstorage.go:[18](https://github.com/Azure/azure-storage-fuse/runs/5639532548?check_suite_focus=true#step:4:18)4: arent ==> aren't
Error: ./component/azstorage/azstorage.go:234: retreived ==> retrieved
Error: ./component/azstorage/azstorage.go:254: Retreived ==> Retrieved
Error: ./component/azstorage/block_blob.go:129: retreive ==> retrieve
Error: ./component/azstorage/block_blob.go:380: Retreive ==> Retrieve
Error: ./component/azstorage/block_blob.go:445: non-existant ==> non-existent
Error: ./component/azstorage/azauthsas.go:73: autheticate ==> authenticate
Error: ./component/azstorage/azauthsas.go:87: autheticate ==> authenticate
Error: ./component/libfuse/libfuse_handler.go:66: convering ==> converting, covering
Error: ./component/libfuse/libfuse_handler.go:135: extnesion ==> extension
Error: ./component/libfuse/libfuse_handler.go:138: retreived ==> retrieved
Error: ./component/libfuse/libfuse_handler.go:148: extnesion ==> extension
Error: ./component/libfuse/libfuse_handler.go:149: extnesion ==> extension
Error: ./component/libfuse/libfuse_handler.go:392: currnet ==> current
Error: ./component/libfuse/libfuse2_handler.go:66: convering ==> converting, covering
Error: ./component/libfuse/libfuse2_handler.go:136: extnesion ==> extension
Error: ./component/libfuse/libfuse2_handler.go:139: retreived ==> retrieved
Error: ./component/libfuse/libfuse2_handler.go:149: extnesion ==> extension
Error: ./component/libfuse/libfuse2_handler.go:150: extnesion ==> extension
Error: ./component/libfuse/libfuse2_handler.go:369: currnet ==> current
Error: ./component/libfuse/libfuse.go:75: childres ==> children
Error: ./component/loopback/loopback_fs_test.go:66: varius ==> various
Error: ./component/loopback/loopback_fs_test.go:66: varius ==> various
Error: ./component/loopback/loopback_fs_test.go:68: varius ==> various
Error: ./component/attr_cache/cacheMap.go:43: cahe ==> cache
Error: ./component/attr_cache/attr_cache.go:150: non-existant ==> non-existent
Error: ./component/attr_cache/attr_cache.go:296: non-existant ==> non-existent
Error: ./azure-pipeline-templates/stress-test.yml:25: cotainer ==> container
Error: ./azure-pipeline-templates/distro-tests.yml:45: strucutre ==> structure
Error: ./internal/base_component.go:42: inteface ==> interface
Error: ./internal/pipeline.go:48: instanciation ==> instantiation
Error: ./internal/attribute.go:81: preseve ==> preserve
Error: ./internal/handlemap/handle_map.go:108: Retreive ==> Retrieve
Error: ./common/stats_collector/stats_collector.go:115: stroage ==> storage
Error: ./common/stats_collector/stats_collector.go:1[22](https://github.com/Azure/azure-storage-fuse/runs/5639532548?check_suite_focus=true#step:4:22): stroage ==> storage
Error: ./common/stats_collector/stats_collector.go:168: stroage ==> storage
Error: ./common/stats_collector/stats.go:72: stroage ==> storage
Error: ./common/stats_collector/stats.go:90: gloabal ==> global
Error: ./common/stats_collector/stats.go:101: stroage ==> storage
Error: ./common/config/keys_tree.go:[55](https://github.com/Azure/azure-storage-fuse/runs/5639532548?check_suite_focus=true#step:4:55): initiliazed ==> initialized
Error: ./common/config/keys_tree.go:222: wil ==> will, well
```

**Comment by gapra-msft** [comment]
Fixed some of these here
https://github.com/Azure/azure-storage-fuse/pull/745

**Review by vibhansa-msft — APPROVED** [review]

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: codespell-project/actions-codespell@master
+        with:
+          check_filenames: true
+          skip: ./.git,./.github/workflows/codespell.yml,.git,*.png,*.jpg,*.svg,*.sum,./vendor,go.sum
```
Comment (vibhansa-msft) [inline .github/workflows/codespell.yml:21]: add ./NOTICE and test_case.csv file in exception list.

Comment (vibhansa-msft) [reply .github/workflows/codespell.yml:21]: ./test/e2e_tests/file_test.go this file as well

Comment (andyzhangx) [reply .github/workflows/codespell.yml]: thanks, I also fixed some spelling issues.

**Review by souravgupta-msft — APPROVED** [review]

**Comment by gapra-msft** [comment]
@andyzhangx not sure why CI is failing here. I think it may have to do with changing the base branch from master to main in your PR? Could you create a fresh PR off main with just your changes?

Alternatively we can just use my PR where I did the same. https://github.com/Azure/azure-storage-fuse/pull/750

**Pull request closed without merging**